### ISSUE TITLE
`json_char` and `strcmp()`

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [json-parser/json-parser#152]
In this piece of code, `name` is of type `json_char *` but `strcmp()` operates on `char *`:
https://github.com/json-parser/json-parser/blob/3ec6608cee3619e4789488ef00503357c06093f0/json.h#L205

Could this be an issue? What are the expected / valid definitions of `json_char`?

**Comment by LB--** [comment]
Good catch. I've personally never been happy with that function taking a C-style string, as it loses a lot of performance by not knowing the length in advance when generally the length can always be known in advance. I'm hoping to eventually replace it with `std::basic_string_view<json_char>` but that is only available in C++17 and above. I'm still mulling over what to do about earlier C++ versions and also backward compatibility.
